pkg/apis/security/types_workloadidentity.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package security
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// WorkloadIdentity is resource that allows Gardener API server to issue JSON Web Tokens.
+// It is designed to be used by components running in the Gardener environment, seed or runtime cluster,
+// that make use of identity federation based on OIDC protocol.
+type WorkloadIdentity struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Spec configures the JSON Web Token issued by the Gardener API server.
+	Spec WorkloadIdentitySpec
+	// Status contain the latest observed status of the WorkloadIdentity.
+	Status WorkloadIdentityStatus
+}
+
+// WorkloadIdentitySpec configures the JSON Web Token issued by the Gardener API server.
+type WorkloadIdentitySpec struct {
+	// Audiences specify the list of OIDC audiences that will be set in the 'aud' claim.
+	Audiences []string
+	// TargetSystem represents specific configurations for the system that will accept the JWTs.
+	TargetSystem TargetSystem
+}
+
+// TargetSystem represents specific configurations for the system that will accept the JWTs.
+type TargetSystem struct {
+	// Type is the type of the target system.
+	Type string
+	// ProviderConfig is the configuration passed to extension resource.
+	ProviderConfig *runtime.RawExtension
```

Comment:
```suggestion
	ProviderConfig runtime.Object
```

See https://github.com/gardener/gardener/blob/0eb9ac3e9820557febacb70164f6267c202d1a67/pkg/apis/core/types_controllerdeployment.go#L24

Reply:
What is the reason for this? I see ControllerDeployment is using `runtime.RawExtension` in the version API, and a lot of other resources also are using `RawExtension` in the internal API instead of `Object`.

Reply:
> What is the reason for this? 

https://github.com/kubernetes/kubernetes/blob/84a7f1ea15e4fcabe88e06ca27de611fa3b74437/staging/src/k8s.io/apimachinery/pkg/runtime/types.go#L53-L96

> and a lot of other resources also are using `RawExtension` in the internal API instead of `Object`.

Yes, because we missed this for some of them back then. The "correct" approach seems to be using `runtime.Object` in the internal version according to the doc string above.